02_commands.html

```diff
@@ -1,206 +1,524 @@
+    <p>The <a>remote end</a> acts as a HTTP server reading requests from the client and writing responses, typically over TCP socket. For the purposes of this specification we model the data transmission between a particular <a>local end</a> and <a>remote end</a> with a <dfn>connection</dfn> object to which the <a>remote end</a> may <dfn>write bytes</dfn> and <dfn>read bytes</dfn>. However the exact details of how this connection works and how it is established are out of scope.</p>
+
+    <p>After such a <a>connection</a> has been established, a remote end MUST run the following steps:</p>
+
+    <ol>
+      <!-- Should mention what happens if it can't be decoded as HTTP -->
+      <!-- Note: the way that fetch is used here isn't quite right because it doesn't
+           specify how to construct a request from network data, or serialize a response. -->
+
+      <li><p><a>Read bytes</a> from the connection until a complete HTTP request can be constructed from the data. Let <var>request</var> be a <a href="https://fetch.spec.whatwg.org/#concept-request" class="externalRef">request</a> object constructed from the received data, according to the requirements of [[!RFC7230]].</p></li>
+
+      <li><p>Let <var>request match</var> be the result of <a>matching a request</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> and <a href="https://fetch.spec.whatwg.org/#concept-request-url" class="externalRef">url</a> as arguments.</p></li>
+
+      <li><p>If <var>request match</var> is of type <a>error</a>, <a>return an error</a> with <var>request match</var>'s <a>error code</a> and jump to step 1.</p>
+
+        <p>Otherwise, let <var>command</var>, <var>session id</var> and <var>element id</var> be <var>request match</var>'s data.</p></li>
+
+      <li><p>If <var>command</var> is not <a>new session</a>:</p>
+        <ol>
+          <li><p>If <var>session id</var> is not equal to the <a>id</a> of any session in the <a>list of active sessions</a>, <a>return an error</a> with <a>error code</a> <a>invalid session id</a>, then jump to step 1.</p>
+
+          <p>Otherwise, let the <a>current session</a> be the session with <a>id</a> <var>session id</var></p>
+          </li>
+        </ol>
+      </li>
+
+      <li><p>If <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> is POST:</p>
+        <ol>
+
+          <li><p>Let <var>success</var> and <var>parameters</var> be the result of <a>parsing as JSON</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-body" class="externalRef">body</a> as the argument.</p></li>
+
+          <li><p>If <var>success</var> is false, or <var>success</var> is true and <var>parameters</var> is not an Object object, <a>return an error</a> with status <a>error code</a> <a>invalid argument</a> and jump back to step 1 in this overall algorithm.</p></li>
```

Comment:
It's unclear to me exactly what `success` indicates here.  Does it tell us if the body could successfully be parsed as JSON?  Perhaps this will be defined in `parsing as JSON`, but I'm curious.

s/Object object/valid JSON object/